frontend/src/host_orchestrator/orchestrator/instancemanager.go

```diff
@@ -365,44 +416,97 @@ func (h *downloadCVDHandler) download() error {
 	return downloadErr
 }
 
-type fetchCVDResult struct {
-	OutDir string
-	Error  error
+// Downloads and organizes the OS artifacts.
+//
+// Artifacts will be organized the next way:
+//  1. $ROOT_DIR/<BUILD_ID>_<TARGET>_full will store a full download.
+//  2. $ROOT_DIR/<BUILD_ID>_<TARGET>_kernel_only will store kernel artifacts only.
+type artifactsManager struct {
+	execContext ExecContext
+	cvdBin      string
+	rootDir     string
+	buildAPI    BuildAPI
+	map_        map[string]*downloadArtifactsMapEntry
+	mapMutex    sync.Mutex
 }
 
-type fetchCVDMapEntry struct {
-	mutex  sync.Mutex
-	result *fetchCVDResult
+func newArtifactsManager(execContext ExecContext, cvdBin, rootDir string, buildAPI BuildAPI) *artifactsManager {
+	return &artifactsManager{
+		execContext: execContext,
+		cvdBin:      cvdBin,
+		rootDir:     rootDir,
+		buildAPI:    buildAPI,
+		map_:        make(map[string]*downloadArtifactsMapEntry),
+	}
 }
 
-type fetchCVDHandler struct {
-	execContext  ExecContext
-	cvdBin       string
-	artifactsDir string
-	map_         map[string]*fetchCVDMapEntry
-	mapMutex     sync.Mutex
+type artifactsBundleType int
+
+const (
+	cvdBundleType artifactsBundleType = iota
+	kernelBundleType
+)
```

Comment:
this type enum should really be two types/functions that do the respective things and share a bunch of code. For example: `artifactsManager` should have an `androidDownloader` and a `kernelDownloader` that both implement `Download`, which in turn is called from `artifactsManager.DownloadAndroidBuild()` and `artifactsManager.DownloadKernelBuild()` respectively.

Reply:
removed the enum and made some changes following the feedback, however didn't create different downloader structs.